src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetDeserializer.cs

```diff
@@ -124,10 +126,133 @@ public virtual IEnumerable ReadResourceSet(ODataResourceSetWrapper resourceSet,
                     Error.Format(SRResources.TypeCannotBeDeserialized, elementType.FullName()));
             }
 
-            foreach (ODataResourceWrapper resourceWrapper in resourceSet.Resources)
+            Collection<object> coll = new Collection<object>();
+
+            if (resourceSet.ResourceSetType == ResourceSetType.ResourceSet)
```

Comment:
Technically, you don't need to special case ResourceSetType.ResourceSet, since line 140-169 will handle a ResourceSet, but it may be more efficient for the common case.

Reply:
For clarify, please add a comment along the lines of the comment above -- that we don't need to special case ResourceSetType.ResourceSet, since the code that handles a deltaresourcesetwrapper will also handle a ResourceSetWrapper, but it may be more efficient for the common case.

---
In reply to: [503428085](https://github.com/OData/WebApi/pull/2275#discussion_r503428085) [](ancestors = 503428085)